include/realizations/catchment/Formulation_Manager.hpp

```diff
@@ -80,56 +79,52 @@ namespace realization {
-
-                        // extract and store layer data from the json
-                        layer_desc.name = "surface layer";
-                        layer_desc.id = 0;
-                        layer_desc.time_step = 3600;
-                        layer_desc.time_step_units = "s";
-
-                        // add the layer to storage
-                        layer_storage.put_layer(layer_desc, layer_desc.id);
-                }
-                else
-                {
+                //Create the default surface layer
+                config::Layer layer;
+                // layer description struct
+                ngen::LayerDescription layer_desc;
+                layer_desc = layer.get_descriptor();
+                // add the default surface layer to storage
+                layer_storage.put_layer(layer_desc, layer_desc.id);
+
+                if(layers_json_array){
+                    
                     for (std::pair<std::string, boost::property_tree::ptree> layer_config : *layers_json_array) 
                     {
-                        // layer description struct
-                        ngen::LayerDescription layer_desc;
-
-                        // extract and store layer data from the json
-                        layer_desc.name = layer_config.second.get<std::string>("name");
-                        layer_desc.id = layer_config.second.get<int>("id");
-                        layer_desc.time_step = layer_config.second.get<double>("time_step");
-                        boost::optional<std::string> layer_units = layer_config.second.get_optional<std::string>("time_step_units");
-                        if (*layer_units == "") layer_units = "s";
-                        layer_desc.time_step_units = *layer_units;
-
-                        // check to see if this layer was allready defined
-                        if (layer_storage.exists(layer_desc.id) )
-                        {
-                            std::string message = "A layer with id = ";
-                            message += std::to_string(layer_desc.id);
-                            message += " was defined more than once";
-
-                            std::runtime_error r_error(message);
-                            
-                            throw r_error;
-                        }
+                        layer = config::Layer(layer_config.second);
+                        layer_desc = layer.get_descriptor();
 
                         // add the layer to storage
                         layer_storage.put_layer(layer_desc, layer_desc.id);
-
                         // debuggin print to see parsed data
-                        std::cout << layer_desc.name << ", " << layer_desc.id << ", " << layer_desc.time_step << ", " << layer_desc.time_step_units << "\n";
+                        //std::cout << layer_desc.name << ", " << layer_desc.id << ", " << layer_desc.time_step << ", " << layer_desc.time_step_units << "\n";
```

Comment:
Should remove code entirely instead of commenting it out.

Reply:
I thought I had most of those cleaned up 🤪 I'll drop it.